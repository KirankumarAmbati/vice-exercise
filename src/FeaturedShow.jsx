import React from 'react';
import PropTypes from 'prop-types';

const FeaturedShow = props => (
  <h1>{props && props.show ? props.show.name : 'Placeholder'}</h1>
);

export default FeaturedShow;

// FeaturedShow.propTypes = {
//   name: PropTypes.string,
// };

// FeaturedShow.defaultProps = {
//   name: 'Placeholder',
// };
