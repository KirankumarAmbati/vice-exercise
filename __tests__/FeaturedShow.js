import React from 'react';
import renderer from 'react-test-renderer';
import FeaturedShow from '../src/components/FeaturedShow';

const testShow = {
  id: 'a1',
  title: 'Gaycation',
  episodes: 24,
  product_image_url: '/images/gaycation.jpg',
};

test('FeaturedShow component', () => {
  const component = renderer.create(<FeaturedShow show={testShow} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
