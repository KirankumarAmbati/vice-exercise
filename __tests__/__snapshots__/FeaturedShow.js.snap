// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedShow component 1`] = `
<div
  className="featured-show"
>
  <img
    alt="Gaycation"
    className="featured-show--image"
    src="/images/gaycation.jpg"
  />
  <h3
    className="featured-show--meta"
  >
    24 episodes
  </h3>
  <h2
    className="featured-show--title"
  >
    Gaycation
  </h2>
</div>
`;
